### PR TITLE
cmake,common: various cleanup for building on MacOS

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@ansiwen FWIW, i am using:
```sh
cmake ..  -DWITH_SYSTEM_BOOST=ON -DWITH_BABELTRACE=OFF -DWITH_BLUESTORE=OFF \
  -DWITH_CCACHE=ON -DWITH_CEPHFS=OFF -DWITH_KRBD=OFF -DWITH_LIBCEPHFS=OFF \
  -DWITH_LTTNG=OFF  -DWITH_LZ4=OFF -DWITH_MANPAGE=ON -DWITH_MGR=OFF \
  -DWITH_MGR_DASHBOARD_FRONTEND=OFF -DWITH_RADOSGW=OFF -DWITH_RDMA=OFF \
  -DWITH_SPDK=OFF -DWITH_SYSTEMD=OFF -DWITH_TESTS=OFF -DWITH_XFS=OFF \
  -DOPENSSL_ROOT_DIR=/usr/local/Cellar/openssl@1.1/1.1.1k
```

along with this patchset.

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2021-04-28_02:38:11-rados-wip-kefu-testing-2021-04-27-2204-distro-basic-smithi/

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -584,7 +584,7 @@ endif(${WITH_LTTNG})
 
 add_subdirectory(global)
 
-if(NOT WIN32)
+if(WITH_CEPHFS)
```
Comment (rzarzynski) [inline src/CMakeLists.txt:592]: :+1: for minimizing the dependencies. The `WITH_CEPHFS && WIN32` case can't happen, right?

**Review by rzarzynski — APPROVED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (tchaikov) [reply src/CMakeLists.txt:592]: see see https://github.com/ceph/ceph/blob/becbe5fc4a72a5ebd97b04e30a1f5c17080dd889/win32_build.sh#L162. moreover, i think we should not hardwire a feature to a platform unless that feature is platform specific.

**Comment by broncotc** [comment]
Thanks for this pr. was doing many of these when trying compiling today.

**Pull request merged**